core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -128,41 +130,52 @@ private[spark] class CompressedMapStatus(
  * @param numNonEmptyBlocks the number of non-empty blocks
  * @param emptyBlocks a bitmap tracking which blocks are empty
  * @param avgSize average size of the non-empty blocks
+ * @param hugeBlockSizesArray sizes of huge blocks by their reduceId.
  */
 private[spark] class HighlyCompressedMapStatus private (
     private[this] var loc: BlockManagerId,
     private[this] var numNonEmptyBlocks: Int,
     private[this] var emptyBlocks: RoaringBitmap,
-    private[this] var avgSize: Long)
+    private[this] var avgSize: Long,
+    private[this] var hugeBlockSizesArray: Array[Tuple2[Int, Byte]])
```

Comment:
Why does hugeBlockSizesArray exist ? Is it for efficient serializable ?
If yes, the then converting it into (Array[Int], Array[Byte]) would be better (with each array written directly - not as Tupe2 object) - more below though.